### ISSUE TITLE
Retain Empty/Blank lines on save? (VSCode)

### ISSUE THREAD

**Issue by MarkMoretto** [mvdan/gofumpt#141]
Not sure if this is a gofumpt thing, but it would be wonderful if I could keep any and all blank lines in a file when saving.  

Currently, I'm adding double forward-slash comment blocks to add spacing, which seems like it should be unnecessary.

If I should post this somewhere else, I'm happy to do so!

---
---

I have the current settings for a dedicated Go workspace in VSCode:

```
{
    "go.inferGopath": false,
    "go.formatTool": "gofumpt",
    "go.lintTool": "golangci-lint",
    "go.buildOnSave": "off",
    "go.lintFlags": [
        "--fast"
    ],
    "go.lintOnSave": "file",
    "editor.formatOnSave": false,
    "[go]": {
        "editor.insertSpaces": false,
        "editor.formatOnSave": false,
        "editor.codeActionsOnSave": {
            "source.organizeImports": true
        },
        "editor.suggest.snippetsPreventQuickSuggestions": false
    }
}
```

Here are my OS/VSCode versions:

+ Version: 1.58.2 (system setup)
+ Commit: c3f126316369cd610563c75b1b1725e0679adfb3
+ Date: 2021-07-14T22:10:15.214Z
+ Electron: 12.0.13
+ Chrome: 89.0.4389.128
+ Node.js: 14.16.0
+ V8: 8.9.255.25-electron.0
+ OS: Windows_NT x64 10.0.18363

Go is version `1.17`.

**Comment by mvdan** [comment]
What do you mean by blank lines? Can you give some examples?

Upstream gofmt, which this tool is based on, gets rid of some of them. This tool is stricter, and removes some more.

**Comment by mvdan** [comment]
Friendly ping :) Still don't understand what you mean.

**Comment by MarkMoretto** [comment]
Hi,

Say you write some code. You have two blank lines between each, I dunno, interface method.

When I save said file, at least one of the blank lines disappear. Along with other "clean up" functions, like removing unused imports. 

I'm sure this is great for production, but I'm just trying to develop programs and would love to not have work/readability disappear during this process. That's all.

tl;dr - I can't easily figure out which setting to toggle to just leave imports/blank lines alone.

**Comment by mvdan** [comment]
> When I save said file, at least one of the blank lines disappear.

This is enforced by gofmt, so I can't do anything about it.

> Along with other "clean up" functions, like removing unused imports.

This is also not done by gofumpt. Depending on your editor, this is done by either gopls or goimports as part of "fix imports".

> tl;dr - I can't easily figure out which setting to toggle to just leave imports/blank lines alone.

Sounds like what you want is to configure your editor to not reformat and fix imports on save. I'm actually not sure how to do this with gopls, as I've never looked, but I imagine it should be possible even if you file a feature request.

In any case, I fear you're not in the right issue tracker - this tool is just one small piece of the puzzle :)

**Comment by MarkMoretto** [comment]
Thanks!

I feel like maybe I'm not on the right language if it's this hard to simply configure.  Guess I'll have to try something else!
